### ISSUE TITLE
Raise note length limit to 2500 characters

### ISSUE THREAD

**Issue by WulfmanDo** [cgeo/cgeo#6591]
##### Detailed steps causing the problem:
-Download  or write  a personal Note  ( with morge the 500x charactars
- edit it, make it longer, what do you want.
- reupload it too the server

##### Actual behavior after performing these steps:

When the note is longer then 500x characters 
c:geo open a popup window and tell you that the personal note will cut off by the geocaching.com server after 500x characters .

##### Expected behavior after performing these steps:

the popup window should only open when he note is longer then 2500x characters  and then to tell you that the note will be cut off by the geocaching.com server after 2500x characters.

##### Version of c:geo used:
2017.04.27-RC
and
2017.05.22

##### Is the problem reproducible for you?
Yes

##### Other comments and remarks:

Groundspeak has made an update to the personal note system and raised the max charakters from 500x  to 2500x charakter.

Is it possible to show the counted charaters of the personalnote field in the top of the edit field ( beside the close/save icon)  like (1478 / 2500)  ?

greetings from germany
Marco / Wulfman_Do

**Comment by pstorch** [comment]
I just looked into it. We have the `PersonalNoteCapability` interface which is implemented by `GCConnector` and `OCApiLiveConnector`, but the 500 character limit is only defined in the `GCConstants` class.
To be clean we would need to define the limits for each connector separately. I would like to add a method in the `PersonalNoteCapability` interface, like `getPersonalNoteMaxChars()`, which needs to be implemented by both connectors.
Then I tried to figure out what the Okapi Limit is. But the [documentation](https://www.opencaching.de/okapi/services/caches/save_personal_notes.html) lacks this information. I tried to upload as much as I could, but stopped at 4416873 bytes.
